crates/node/src/scheduler/mod.rs

```diff
@@ -310,7 +310,9 @@ impl Scheduler {
                         state.task_queue.remove(&task.tx);
                     }
 
-                    self.reschedule(&task).await?;
```

Comment:
`self.state` is locked _here_, but `fn reschedule()` tried to lock it again, in order to push the task back to queue.